### PR TITLE
MOTOR-941 Bump pymongo requirement to 4

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Let's update the [requirements page](https://motor.readthedocs.io/en/stable/requirements.html) and [README](https://github.com/mongodb/motor#dependencies) too.

File: doc/requirements.rst [hunk 1]
```diff
@@ -47,6 +47,8 @@ Motor and PyMongo
 +-------------------+-----------------+
 | 2.5               | 3.12+           |
 +-------------------+-----------------+
+| 3.0               | 4.0+            |
++-------------------+-----------------+
 
 Motor and MongoDB
```
Comment (ShaneHarvey) [inline doc/requirements.rst:53]: Let's add 3.0 to this "Motor and MongoDB" table too.

File: doc/requirements.rst [hunk 2]
```diff
@@ -124,31 +128,33 @@ Motor 1.2.5 and 1.3.1 add compatibility with Python 3.7, but at the cost of
...
-+---------------+-----+-----+-----+-----+-----+-----+-------+-------+-----+-----+-----+-----+
-|               | 2.2 |**N**|**N**|**N**|**N**|**N**|**N**  |  Y    |  Y  |  Y  |  Y  |**N**|
-+---------------+-----+-----+-----+-----+-----+-----+-------+-------+-----+-----+-----+-----+
-|               | 2.3 |**N**|**N**|**N**|**N**|**N**|**N**  |  Y    |  Y  |  Y  |  Y  |**N**|
-+---------------+-----+-----+-----+-----+-----+-----+-------+-------+-----+-----+-----+-----+
-|               | 2.4 |**N**|**N**|**N**|**N**|**N**|**N**  |  Y    |  Y  |  Y  |  Y  |  Y  |
-+---------------+-----+-----+-----+-----+-----+-----+-------+-------+-----+-----+-----+-----+
-|               | 2.5 |**N**|**N**|**N**|**N**|**N**|**N**  |  Y    |  Y  |  Y  |  Y  |  Y  |
-+---------------+-----+-----+-----+-----+-----+-----+-------+-------+-----+-----+-----+-----+
++-------------------------------------------------------------------------------------+
```
Comment (ShaneHarvey) [inline doc/requirements.rst:131]: This table doesn't render correctly anymore: https://motor--161.org.readthedocs.build/en/161/requirements.html#motor-and-python

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
https://motor--161.org.readthedocs.build/en/161/installation.html#dependencies
> Python 3.5+

Let's update this to 3.7+

File: doc/requirements.rst (hunk 1 above)
Comment (blink1073) [reply doc/requirements.rst:53]: Done

File: doc/requirements.rst (hunk 2 above)
Comment (blink1073) [reply doc/requirements.rst:138]: Fixed

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**